### PR TITLE
Issue/passport oauth2

### PR DESCRIPTION
**Pull request by labajo**
Solve #22

### CODE REVIEW

File: portal/lib/config/config.json [hunk 1]
```diff
@@ -8,12 +8,23 @@
...
-    "scope": ["read:org", "write:repo_hook"],
-    "callbackURL": "http://localhost:9000/auth/callback"
+  "oauth2Strategy": {
+    "authorizationURL": "http://localhost:3000/oauth/authorize",
+    "tokenURL": "http://localhost:3000/oauth/token",
+    "clientID": "ca9b0a42-0392-4dd1-8a47-487860cb617b",
+    "clientSecret": "9c03033b-5a60-4304-b30c-95babb009925",
+    "callbackURL": "http://localhost:9000/auth/callback",
+    "authorizeParams": {
+      //"prompt": "login"
```
Comment (labajo) [inline portal/lib/config/config.json]: Si se descomenta, siempre se pedirá el user, pass en accounts.

File: portal/lib/config/config.json [hunk 2]
```diff
@@ -8,12 +8,23 @@
...
-    "clientSecret": "65a91c4eda899d08ab89a297cf70dee3168368d3",
-    "scope": ["read:org", "write:repo_hook"],
-    "callbackURL": "http://localhost:9000/auth/callback"
+  "oauth2Strategy": {
+    "authorizationURL": "http://localhost:3000/oauth/authorize",
+    "tokenURL": "http://localhost:3000/oauth/token",
+    "clientID": "ca9b0a42-0392-4dd1-8a47-487860cb617b",
+    "clientSecret": "9c03033b-5a60-4304-b30c-95babb009925",
+    "callbackURL": "http://localhost:9000/auth/callback",
+    "authorizeParams": {
```
Comment (dlozlla) [inline portal/lib/config/config.json]: are these additional params for the authorization request?

File: portal/lib/config/config.json [hunk 3]
```diff
@@ -8,12 +8,23 @@
...
+      //"prompt": "login"
+    },
+    "scope": ["profile"],
+    "customHeaders": {
+      "Authorization": "Basic Y2E5YjBhNDItMDM5Mi00ZGQxLThhNDctNDg3ODYwY2I2MTdiOjljMDMwMzNiLTVhNjAtNDMwNC1iMzBjLTk1YmFiYjAwOTkyNQ=="
```
Comment (labajo) [inline portal/lib/config/config.json]: Esta cabecera se puede calcular desde la lógica usando el clientID and clientSecret. Pero manchará el código por simplificar esta config.

File: portal/lib/config/config.json [hunk 4]
```diff
@@ -8,12 +8,23 @@
...
+    "authorizationURL": "http://localhost:3000/oauth/authorize",
+    "tokenURL": "http://localhost:3000/oauth/token",
+    "clientID": "ca9b0a42-0392-4dd1-8a47-487860cb617b",
+    "clientSecret": "9c03033b-5a60-4304-b30c-95babb009925",
+    "callbackURL": "http://localhost:9000/auth/callback",
+    "authorizeParams": {
+      //"prompt": "login"
+    },
+    "scope": ["profile"],
+    "customHeaders": {
```
Comment (dlozlla) [inline portal/lib/config/config.json]: why do we need this header? the library cannot build the standard request to token endpoint based on clientId and secret?

File: portal/lib/config/config.json (hunk 2 above)
Comment (labajo) [inline portal/lib/config/config.json]: Yes, for the authserver /oauth/authorize endpoint

File: portal/lib/config/config.json (hunk 3 above)
Comment (dlozlla) [inline portal/lib/config/config.json]: why then asking for cliend_id  and secret in the config above.... you have to update that in two places? I though the passport library would calculate the authorization header for us, so we don't need to do anything at our code. That's not the case?

File: portal/lib/tdaf-box.js [hunk 5]
```diff
@@ -28,7 +28,7 @@ var express = require('express'),
     path = require('path'),
```
Comment (dlozlla) [inline portal/lib/tdaf-box.js]: we should rename the file to druida.js

File: portal/lib/config/config.json (hunk 4 above)
Comment (labajo) [inline portal/lib/config/config.json]: To get an access_token using a code our authserver uses authorization Basic b64(clientId:clientSecret), however passport-oauth2 includes clientId and clientSecret in the body of the request. This way is not supported by authserver yet, but it is included in oauth2 spec.

File: portal/public/tdafbox.css [hunk 6]
```diff
@@ -239,7 +239,7 @@ div.modal-footer > button {
     width: 75px;
```
Comment (dlozlla) [inline portal/public/tdafbox.css]: we should rename the file to druida.css

Comment (labajo) [inline portal/public/tdafbox.css]: I think is better create a new issue to change the old name references.

File: portal/lib/config/config.json (hunk 1 above)
Comment (dlozlla) [inline portal/lib/config/config.json]: OK

File: portal/lib/config/config.json (hunk 2 above)
Comment (dlozlla) [inline portal/lib/config/config.json]: OK

File: portal/lib/config/config.json (hunk 4 above)
Comment (dlozlla) [inline portal/lib/config/config.json]: OK, I see. Then I agree to keep this config here and simply remove it in the future when we enhance the authserver

File: portal/public/tdafbox.css (hunk 6 above)
Comment (dlozlla) [inline portal/public/tdafbox.css]: OK, simply rename the previous one to druida-portal.js

**Comment by dlozlla** [comment]
LGTM

**Pull request merged**